qemu/tests/qemu_guest_agent.py

```diff
@@ -885,6 +893,100 @@ def gagent_check_fsfreeze(self, test, params, env):
         # Finally, do something after thaw.
         self._action_after_fsthaw(test, params, env)
 
+    @error_context.context_aware
+    def gagent_check_hotplug_virtio_console(self, test, params, env):
+        """
+        hotplug guest agent device
+
+        :param test: kvm test object
+        :param params: Dictionary with the test parameters
+        :param env: Dictionary with test environment.
+        """
+        def cmd_qmp_log(vm, cmd, args):
+            """
+            log the qmp command and its output
+
+            :param vm: VM object
+            :param cmd: the qmp command
+            :param args: the parameters of qmp command
+            """
+            logging.debug("[qmp cmd %s] %s" % (cmd, args))
+            reply = vm.monitor.cmd_qmp(cmd, args)
+            logging.debug("[qmp reply] %s" % reply)
+            if "error" in reply:
+                if reply["error"]["class"] == "CommandNotFound":
+                    raise error.TestNAError("qmp command %s not supported"
+                                            % cmd)
+                else:
+                    raise error.TestFail("qmp error: %s" % reply["error"]["desc"])
+            return reply
+
+        def pci_serial_add(vm, name, chardev):
```

Comment:
Maybe you can use qcontainer.simple_hotplug() instead, if no extra process besides hotplug